### PR TITLE
Do not include auth details in `conda-meta/*.json` files by default

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Related to https://github.com/conda/conda/pull/13936.

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/core/prefix_data.py [hunk 1]
```diff
@@ -112,7 +113,7 @@ def _get_json_fn(self, prefix_record):
             )
         return fn + ".json"
 
-    def insert(self, prefix_record):
+    def insert(self, prefix_record, dump_auth=False):
```
Comment (travishathaway) [inline conda/core/prefix_data.py:116]:
```suggestion
    def insert(self, prefix_record, include_tokens=False):
```

I think this is a little more descriptive.

File: conda/core/prefix_data.py [hunk 2]
```diff
@@ -132,8 +133,14 @@ def insert(self, prefix_record):
                 context,
             )
             rm_rf(prefix_record_json_path)
-
-        write_as_json_to_file(prefix_record_json_path, prefix_record)
+        if dump_auth:
```
Comment (travishathaway) [inline conda/core/prefix_data.py:136]:
```suggestion
        if include_tokens:
```

**Review by travishathaway — COMMENTED** [review]
Left a suggestion for a slightly better variable name.

I was curious though. Are we okay with making this default behavior? Is this the type of thing that gets to bypass the deprecation notices?

**Comment by jaimergp** [comment]
There was no option before and I would say this is a security oversight. I'm not aware of any conversations discussing this in the past, but I didn't run a thorough investigation. 

The risk here is that the anaconda-client tokens, which should / could be protected by the OS permission system under `.continuum/anaconda-client`, are being exported silently to a location that might not benefitting from the same security treatment. I think that constitutes a bug, not a "behavior".

File: conda/core/prefix_data.py (hunk 1 above)
Comment (jezdez) [reply conda/core/prefix_data.py:116]: To stay consistent with the argument added in (my review of) #13936:

```suggestion
    def insert(self, prefix_record, remove_auth=True):
```

File: conda/core/prefix_data.py [hunk 3]
```diff
@@ -132,8 +133,14 @@ def insert(self, prefix_record):
...
+            write_as_json_to_file(prefix_record_json_path, prefix_record)
+        else:
+            dumped_prefix_record = prefix_record.dump()
+            dumped_prefix_record["url"] = remove_auth(
+                mask_anaconda_token(prefix_record.url)
+            )
+            write_as_json_to_file(prefix_record_json_path, dumped_prefix_record)
```
Comment (jezdez) [inline conda/core/prefix_data.py:143]:
```suggestion
        if remove_auth:
            prefix_record_json = prefix_record.dump()
            prefix_record_json["url"] = remove_auth(
                mask_anaconda_token(prefix_record.url)
            )
        else:
            prefix_record_json = prefix_record
        write_as_json_to_file(prefix_record_json_path, prefix_record_json)
```

**Review by jezdez — DISMISSED** [review]
Approved with comments

**Comment by jaimergp** [comment]
pre-commit.ci autofix

**Comment by jaimergp** [comment]
> Approved with comments

Comments applied, requesting reapproval :D

**Review by travishathaway — APPROVED** [review]
I checked with the Anaconda Navigator team on this, and they do not foresee any difficulties with such a change. Therefore, I'm happy to approve it now 😄 🚀

**Pull request merged**